app/main/views/services.py

```diff
@@ -503,6 +503,62 @@ def update_section_submission(framework_slug, lot_slug, service_id, section_id,
                                 service_id=service_id))
 
 
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/<service_id>/remove/<question_slug>',
+            methods=['GET', 'POST'])
+@login_required
+def remove_subsection(framework_slug, lot_slug, service_id, question_slug):
+    if request.args.get("confirm") and request.method == "POST":
+        # Remove the section
+        try:
+            draft = data_api_client.get_draft_service(service_id)['services']
+        except HTTPError as e:
+            abort(e.status_code)
+
+        if not is_service_associated_with_supplier(draft):
+            abort(404)
+
+        content = content_loader.get_manifest(framework_slug, 'edit_submission').filter(draft)
+        question_to_remove = content.get_question_by_slug(question_slug)
+        containing_section = get_containing_section(content, question_to_remove)
+        fields_to_remove = question_to_remove.form_fields
+        section_responses = [field for field in containing_section.get_field_names() if field in draft]
+        fields_remaining_after_removal = [field for field in section_responses if field not in fields_to_remove]
+
+        if len(fields_remaining_after_removal) > 0:
```

Comment:
I don't think we really need these checks here – it's easier to send the request and let the API return an error if "at leas one" validation fails. This will require us to handle the API errors, but gets rid of the validation logic duplication (that can get more complicated depending on the section). We also don't really care about removing last section on a not-submitted draft.